tests/core/test_impersonation_tests.py

```diff
@@ -84,15 +113,48 @@ def create_user(check_original_docker_image):
                 f"{command!r} without a password prompt (check sudoers file)?\n"
                 f"{e.stdout.decode() if e.stdout else ''}"
             )
-    yield
+    yield TEST_USER
     subprocess.check_call(["sudo", "userdel", "-r", TEST_USER])
 
 
+@pytest.fixture
+def create_airflow_home(create_user, tmp_path, monkeypatch):
+    sql_alchemy_conn = conf.get_mandatory_value("database", "sql_alchemy_conn")
+    username = create_user
+    airflow_home = tmp_path / "airflow-home"
+
+    if not airflow_home.exists():
+        airflow_home.mkdir(parents=True, exist_ok=True)
+
+    permissions = {
+        # By default, ``tmp_path`` save temporary files/directories in user temporary directory
+        # something like: `/tmp/pytest-of-root` and other users might be limited to access to it,
+        # so we need to grant at least r/o access to everyone.
+        airflow_home: 0o777,
+        # Grant full access to system-wide temporary directory (if for some reason it not set)
+        tempfile.gettempdir(): 0o777,
+    }
+
+    if sql_alchemy_conn.lower().startswith("sqlite"):
+        sqlite_file = Path(sql_alchemy_conn.replace("sqlite:///", ""))
+        # Grant r/w access to sqlite file.
+        permissions[sqlite_file] = 0o766
+        # Grant r/w access to directory where sqlite file placed
+        # otherwise we can't create temporary files during write access.
+        permissions[sqlite_file.parent] = 0o777
```

Comment:
This just dummy solution, we grant r/w access to `/root/airflow/sqlite`

Reply:
Well. It's not as dummy. It looks like it will restore original permissions with context manager, so that one LGTM.